### PR TITLE
Adding new issue template

### PR DESCRIPTION
**Pull request by zendranm**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

Solves issue #1204 

### Describe your change:

- [ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [x] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new JavaScript files are placed inside an existing directory.
- [ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1206"></a>

### CODE REVIEW

**Review by appgurueu — COMMENTED** [review]
The feature request template is not really applicable; algorithms are not to be requested (rather, PRs should be opened directly). The bug template should probably have a "bug" label. Hints to the one opening the issue such as "Before you open an issue ..." should be comments.

**Comment by Panquesito7** [comment]
> The feature request template is not really applicable...

I still think we should keep the feature request template. For example: what if I want to add a new workflow or functionality and make an issue before making any PRs?
@raklaptudirm, what do you think about this?

File: .github/ISSUE_TEMPLATE/bug-issue-template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug-issue-template.md:1]: I believe we should use the new [**GitHub issue forms**](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). They are way better and much easier to use, IMO.
Should be pretty easy to port this one. If you need any help, let us know here on our [**Discord server**](https://the-algorithms.com/discord/). 🙂

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by raklaptudirm** [comment]
> > The feature request template is not really applicable...
> 
> 
> 
> I still think we should keep the feature request template. For example: what if I want to add a new workflow or functionality and make an issue before making any PRs?
> 
> @raklaptudirm, what do you think about this?
> 
> 

We should explicitly mention in the issue template that it is not for requesting algorithms imo.

File: .github/ISSUE_TEMPLATE/bug-issue-template.md (hunk 1 above)
Comment (zendranm) [reply .github/ISSUE_TEMPLATE/bug-issue-template.md]: @Panquesito7 that is a good point. I've converted the bug issue template to `yaml` format and added a template for the feature request.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+labels: ["bug"]
 body:
   - type: markdown
     attributes:
-      value: 'Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md).'
+      value: "###Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md)."
+   - type: textarea
+    attributes:
+      label: Description
+      description: Explain what is the problem.
```
Comment (appgurueu) [inline .github/ISSUE_TEMPLATE/bug_report.yml:12]: what the problem is

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-description: Feature request template.
-title: '[Feature]: '
-labels: ['feature']
+description:  "Suggest features, propose improvements, discuss new ideas"
+title: '[FEATURE]: '
+labels: ['enhancement']
 body:
 - type: markdown
     attributes:
       value: '## This issue template is not for requesting new algorithms. For new algorithms, PRs should be opened directly.'
```
Comment (appgurueu) [inline .github/ISSUE_TEMPLATE/feature_request.yml:8]: This is dirty.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   - type: markdown
     attributes:
-      value: 'Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md).'
+      value: '###Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md).'
```
Comment (appgurueu) [inline .github/ISSUE_TEMPLATE/feature_request.yml:11]: Same here.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 3 above)
Comment (zendranm) [reply .github/ISSUE_TEMPLATE/feature_request.yml:8]: @appgurueu What do you mean by "dirty"?

Comment (appgurueu) [reply .github/ISSUE_TEMPLATE/feature_request.yml:8]: You're using a Markdown field - which will often end up being submitted as part of the feature request - just to inform the user of something.

Comment (zendranm) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: I am not sure I see the problem. According to [the documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#markdown):

> You can use a markdown element to display Markdown in your form that provides extra context to the user, but is not submitted.

So it seems like a perfect match for this use case. @appgurueu

Comment (Panquesito7) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: > You're using a Markdown field - which will often end up being submitted as part of the feature request - just to inform the user of something.

Nope. That is removed once the issue is submitted and is only available to the user _before_ submitting.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: Bug report
+description: "Create a report to help us improve"
+title: "[BUG]: "
+labels: ["bug"]
+body:
+    - type: markdown
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:6]:
```suggestion
   - type: markdown
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+name: Bug report
+description: "Create a report to help us improve"
+title: "[BUG]: "
+labels: ["bug"]
+body:
+   - type: markdown
+    attributes:
+      value: "### Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md)."
+   - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]: I believe we need to set an ID name for each of the `textarea` fields.
```suggestion
   - type: textarea
    id: description
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    attributes:
+      label: Description
+      description: Explain what the problem is.
+    validations:
+      required: true
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:15]:
```suggestion
  - type: textarea
    id: exceptedbhv
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    attributes:
+      label: Expected Behavior
+      description: Describe what was the expected behavior.
+    validations:
+      required: true
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:21]:
```suggestion
  - type: textarea
    id: actualbhv
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    attributes:
+      label: Actual Behavior
+      description: Describe what actually happens.
+    validations:
+      required: true
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:27]:
```suggestion
  - type: textarea
    id: steps
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+      label: Steps to reproduce (if applicable)
+      description: List steps to reproduce the behavior.
+      placeholder: |
+        1. 
+        2. 
+        3. 
+        4.
+    validations:
+      required: false
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:38]:
```suggestion
  - type: textarea
    id: specifications
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+description: 'Suggest features, propose improvements, discuss new ideas'
+title: '[FEATURE]: '
+labels: ['enhancement']
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ## This issue template is not for requesting new algorithms. For new algorithms, PRs should be opened directly.
+        ## Make sure your issue isn't a duplicate and you follow our [contributing guidelines](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md)
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/feature_request.yml:11]:
```suggestion
  - type: textarea
    id: description
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    attributes:
+      label: Motivation
+      description: Describe what is the motivation behind this feature.
+    validations:
+      required: true
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/feature_request.yml:17]:
```suggestion
  - type: textarea
    id: examples
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    attributes:
+      label: Examples
+      description: If possible, provide examples of how this feature can be used.
+    validations:
+      required: false
+  - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/feature_request.yml:23]:
```suggestion
  - type: textarea
    id: workarounds
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 14]
```diff
@@ -0,0 +1,52 @@
+name: Bug report
+description: "Create a report to help us improve"
+title: "[BUG]: "
+labels: ["bug"]
+body:
+   - type: markdown
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:6]:
```suggestion
  - type: markdown
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    attributes:
+      value: "### Before you open an issue, please verify if a similar one already exists or has been closed before. More details about the process of contributing can be found in [CONTRIBUTING.md](https://github.com/TheAlgorithms/JavaScript/blob/master/CONTRIBUTING.md)."
+   - type: textarea
```
Comment (Panquesito7) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]:
```suggestion
  - type: textarea
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Wrong indentation.

**Review by Panquesito7 — APPROVED** [review]
Looks great. Thanks! 🚀

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+        2. 
+        3. 
+        4.
+    validations:
+      required: false
+  - type: textarea
+    id: specifications
+    attributes:
+      label: Specifications
+      description: Specify version, platform, and web browser (if applicable).
```
Comment (appgurueu) [inline .github/ISSUE_TEMPLATE/bug_report.yml:46]: This doesn't really make sense. This repo is not for a browser; it's for Node.js. The Node.js version is *very unlikely* to be the cause of an issue.

Comment (zendranm) [reply .github/ISSUE_TEMPLATE/bug_report.yml:46]: @appgurueu Which specification details would be useful then? Or should I remove this section entirely?

**Comment by zendranm** [comment]
@appgurueu Since there is not much time until the end of the Hactoberfest, would you mind adding the `hacktoberfest-accepted` label to this PR? I will add any required code-review changes once I get a [response here](https://github.com/TheAlgorithms/JavaScript/pull/1206#discussion_r1008275683).

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 16 above)
Comment (zendranm) [reply .github/ISSUE_TEMPLATE/bug_report.yml:46]: @appgurueu Regarding my previous comment, can you explain what the `Specifications` section should look like?

Comment (appgurueu) [reply .github/ISSUE_TEMPLATE/bug_report.yml:46]: I'd remove it entirely.

**Review by Panquesito7 — APPROVED** [review]
Great, thanks! 🚀

**Review by appgurueu — APPROVED** [review]

**Pull request merged**